source/dxgi/dxgi_swapchain.cpp

```diff
@@ -464,6 +434,7 @@ HRESULT STDMETHODCALLTYPE DXGISwapChain::ResizeBuffers1(UINT BufferCount, UINT W
 			break;
 		case 11:
 			assert(_runtime != nullptr);
+			perform_callcounter_cleanup();
```

Comment:
I don't like that this D3D11 specific code is part of the DXGI hooks. This should be generalized. I'm not sure it's even necessary, since `D3D11Device::Release` repeats this work.

Reply:
I don't see where D3D11Device::Release is called when the buffers are resized? Debugging shows when resizing the screen, only `DXGI::ResizeBuffers` is called, not the Release in D3D11Device. So the cleanup has to take place at different places. The callcounter cleanup here is done to prevent memory leaks when the screen is resized. It's the location to make sure the call counters in at least the immediatecontext are cleared (and their references).

Reply:
Alright. I can think of a solution later. It's fine for the initial merge.